src/cgame/cg_rocket_draw.cpp

```diff
@@ -2174,6 +2174,131 @@ class PredictedMineEfficiencyElement : public HudElement
 	std::unordered_map<int, std::string> pluralSuffix;
 };
 
+class BarbsHudElement : public HudElement
+{
+public:
+	BarbsHudElement ( const Rocket::Core::String& tag ) :
+	HudElement ( tag, ELEMENT_ALIENS ),
+	numBarbs( 0 ),
+	maxBarbs( 0 ),
+	regenerationInterval ( 0 ),
+	t0 ( 0 ),
+	offset ( 0 ) {}
+
+	void OnAttributeChange( const Rocket::Core::AttributeNameList& changed_attributes )
+	{
+		HudElement::OnAttributeChange( changed_attributes );
+		if ( changed_attributes.find( "src" ) != changed_attributes.end() )
+		{
+			maxBarbs = BG_Weapon( WP_ALEVEL3_UPG )->maxAmmo;
+			if ( maxBarbs > 0 )
+			{
+				src = GetAttribute<Rocket::Core::String>( "src", "" );
+				Rocket::Core::String base( va("<img class='barbs' src='%s' />", src.CString() ) );
+				Rocket::Core::String rml;
+
+				for ( int i = 0; i < maxBarbs; i++ )
+				{
+					rml += base;
+				}
+
+				// SetInnerRML deletes all child Elements => bring children up to date
+				children.clear();
+				SetInnerRML( rml );
+				for( int i = 0; i < maxBarbs; i++) {
+					children.push_back( GetChild( i ) );
```

Comment:
If the number of children will always be consistent with the number of barbs then I would argue to just use `GetChild` and not manage another data structure that does basically the same. (Not mainly for performance reasons but to keep the code small and readable.)

If you can think of us adding non-barb children at some point which would potentially break the code then stick with this but justify it with the variable name, i.e. call it `barbs` or `barbChildren` instead of `children`, because then the point is to have a structure containing *specific* children.